### ISSUE TITLE
OpenAPI file standard indication incorrect: change swagger: "3.0" to openapi: "3.0.0"

### ISSUE THREAD

**Issue by DSotnikov** [frikky/security-openapis#22]
These files actually do not follow the OpenAPI Specification standard.

swagger as the keyword was used for version 2.0.

If you are using the 3.0 version of the standard, start you files with: **openapi**: "3.0.0"

See https://swagger.io/docs/specification/basic-structure/ for details

**Comment by frikky** [comment]
You're right, thanks! I believe this has to do with us focusing on Swagger when we got started, then made the translator for >=v3 without fixing every part of the structure.

**Comment by NobbZ** [comment]
As we currently need a OpenAPI spec for Zammad we checked this repository.

As our tooling failed due to the invalid `swagger: "3.0"`, we tried setting `swagger: "2.0"` as well as `openapi: 3.0.0`, in both cases errors have been reported.

So whatever the current state is, it is neither Swagger, nor OpenAPI.

Is there any ETA when this will get fixed?

**Comment by frikky** [comment]
> As we currently need a OpenAPI spec for Zammad we checked this repository.
> 
> As our tooling failed due to the invalid `swagger: "3.0"`, we tried setting `swagger: "2.0"` as well as `openapi: 3.0.0`, in both cases errors have been reported.
> 
> So whatever the current state is, it is neither Swagger, nor OpenAPI.
> 
> Is there any ETA when this will get fixed?

Hey! What were the issue you found? 

The change here would be to just change it to 
```
openapi: 3.0.0
```

**Comment by NobbZ** [comment]
If I do that, then the editor.swagger.io still complaints about structural and semantic errors.

<details><summary>With `openapi: 3.0.0`</summary>
> Structural error at 
> should NOT have additional properties
> additionalProperty: host, basePath, schemes
> Jump to line 0
> Structural error at paths./tickets.post.parameters.0
> should NOT have additional properties
> additionalProperty: multiline
> Jump to line 35
> Structural error at paths./tickets.post.parameters.0.in
> should be equal to one of the allowed values
> allowedValues: path, query, header, cookie
> Jump to line 35
> Structural error at paths./tickets.post.parameters.1
> should NOT have additional properties
> additionalProperty: multiline
> Jump to line 45
> Semantic error at paths./tickets.get.requestBody
> GET operations cannot have a requestBody.
> Jump to line 74
> Structural error at paths./tickets/search.get.parameters.3
> should NOT have additional properties
> additionalProperty: multiline
> Jump to line 108
> Semantic error at paths./tickets/search.get.requestBody
> GET operations cannot have a requestBody.
> Jump to line 116
> Structural error at paths./users.post.parameters.0
> should NOT have additional properties
> additionalProperty: multiline
> Jump to line 132
> Structural error at paths./users.post.parameters.0.in
> should be equal to one of the allowed values
> allowedValues: path, query, header, cookie
> Jump to line 132
> Structural error at paths./users.post.parameters.1
> should NOT have additional properties
> additionalProperty: multiline
> Jump to line 140
> Structural error at paths./users/search.get.parameters.3
> should NOT have additional properties
> additionalProperty: multiline
> Jump to line 186
> Semantic error at paths./users/search.get.requestBody
> GET operations cannot have a requestBody.
> Jump to line 194
> Semantic error at paths./tickets/{id}.get.requestBody
> GET operations cannot have a requestBody.
> Jump to line 216
> Semantic error at paths./tickets/{id}.delete.requestBody
> DELETE operations cannot have a requestBody.
> Jump to line 237
</details>

If though I use `swagger: "2.0"` I do get a different set of errors:

<details><summary>With `swagger: "2.0"`</summary>
> Structural error at 
> should NOT have additional properties
> additionalProperty: servers, components
> Jump to line 0
> Structural error at schemes.0
> should be equal to one of the allowed values
> allowedValues: http, https, ws, wss
> Jump to line 19
> Structural error at paths./tickets.post
> should NOT have additional properties
> additionalProperty: requestBody
> Jump to line 22
> Structural error at paths./tickets.post.responses.default
> should NOT have additional properties
> additionalProperty: content
> Jump to line 24
> Structural error at paths./tickets.post.parameters.0
> should NOT have additional properties
> additionalProperty: multiline, example
> Jump to line 35
> Structural error at paths./tickets.post.parameters.1
> should NOT have additional properties
> additionalProperty: multiline, example, schema
> Jump to line 45
> Structural error at paths./tickets.get
> should NOT have additional properties
> additionalProperty: requestBody
> Jump to line 61
> Structural error at paths./tickets.get.responses.default
> should NOT have additional properties
> additionalProperty: content
> Jump to line 63
> Structural error at paths./tickets/search.get
> should NOT have additional properties
> additionalProperty: requestBody
> Jump to line 77
> Structural error at paths./tickets/search.get.responses.default
> should NOT have additional properties
> additionalProperty: content
> Jump to line 79
> Structural error at paths./tickets/search.get.parameters.0
> should NOT have additional properties
> additionalProperty: schema
> Jump to line 90
> Structural error at paths./tickets/search.get.parameters.0
> should have required property 'type'
> missingProperty: type
> Jump to line 90
> Structural error at paths./tickets/search.get.parameters.1
> should NOT have additional properties
> additionalProperty: schema
> Jump to line 96
> Structural error at paths./tickets/search.get.parameters.1
> should have required property 'type'
> missingProperty: type
> Jump to line 96
> Structural error at paths./tickets/search.get.parameters.2
> should NOT have additional properties
> additionalProperty: schema
> Jump to line 102
> Structural error at paths./tickets/search.get.parameters.2
> should have required property 'type'
> missingProperty: type
> Jump to line 102
> Structural error at paths./tickets/search.get.parameters.3
> should NOT have additional properties
> additionalProperty: multiline, example, schema
> Jump to line 108
> Structural error at paths./users.post
> should NOT have additional properties
> additionalProperty: requestBody
> Jump to line 119
> Structural error at paths./users.post.responses.default
> should NOT have additional properties
> additionalProperty: content
> Jump to line 121
> Structural error at paths./users.post.parameters.0
> should NOT have additional properties
> additionalProperty: multiline, example
> Jump to line 132
> Structural error at paths./users.post.parameters.1
> should NOT have additional properties
> additionalProperty: multiline, example, schema
> Jump to line 140
> Structural error at paths./users/search.get
> should NOT have additional properties
> additionalProperty: requestBody
> Jump to line 155
> Structural error at paths./users/search.get.responses.default
> should NOT have additional properties
> additionalProperty: content
> Jump to line 157
> Structural error at paths./users/search.get.parameters.0
> should NOT have additional properties
> additionalProperty: schema
> Jump to line 168
> Structural error at paths./users/search.get.parameters.0
> should have required property 'type'
> missingProperty: type
> Jump to line 168
> Structural error at paths./users/search.get.parameters.1
> should NOT have additional properties
> additionalProperty: schema
> Jump to line 174
> Structural error at paths./users/search.get.parameters.1
> should have required property 'type'
> missingProperty: type
> Jump to line 174
> Structural error at paths./users/search.get.parameters.2
> should NOT have additional properties
> additionalProperty: schema
> Jump to line 180
> Structural error at paths./users/search.get.parameters.2
> should have required property 'type'
> missingProperty: type
> Jump to line 180
> Structural error at paths./users/search.get.parameters.3
> should NOT have additional properties
> additionalProperty: multiline, example, schema
> Jump to line 186
> Structural error at paths./tickets/{id}.get
> should NOT have additional properties
> additionalProperty: requestBody
> Jump to line 197
> Structural error at paths./tickets/{id}.get.responses.default
> should NOT have additional properties
> additionalProperty: content
> Jump to line 199
> Structural error at paths./tickets/{id}.get.parameters.0
> should NOT have additional properties
> additionalProperty: schema
> Jump to line 210
> Structural error at paths./tickets/{id}.get.parameters.0
> should have required property 'type'
> missingProperty: type
> Jump to line 210
> Structural error at paths./tickets/{id}.delete
> should NOT have additional properties
> additionalProperty: requestBody
> Jump to line 218
> Structural error at paths./tickets/{id}.delete.responses.default
> should NOT have additional properties
> additionalProperty: content
> Jump to line 220
> Structural error at paths./tickets/{id}.delete.parameters.0
> should NOT have additional properties
> additionalProperty: schema
> Jump to line 231
> Structural error at paths./tickets/{id}.delete.parameters.0
> should have required property 'type'
> missingProperty: type
> Jump to line 231
</details>

On top of that, even if the version was specified correctly, we were unable to use it, as all responses are handwaved as strings…

Without proper declaration of the responses we can not use microcks to mock responses, we can not use a generic openapi client, we can not use it to verify responses on the firewall.

**Comment by frikky** [comment]
Thanks for the clarification - I'm moving this over to the main repo, as it's a code generation issue, handled by the backend.

That way it will get more attention too ;)
